lib/elliptic/eddsa/key.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+var elliptic = require('../../elliptic');
+var utils = elliptic.utils;
+var assert = utils.assert;
+var lazyComputed = utils.lazyComputed;
+
+/**
+* @param {EDDSA} eddsa - instance
+* @param {Object} params - public/private key parameters
+*
+* @param {Array<Byte>} [params.secret] - secret seed bytes
+* @param {bn} [params.priv] - private key scalar (aka `a` in eddsa terms)
+* @param {Point} [params.pub] - public key point (aka `A` in eddsa terms)
+* @param {Array<Byte>} [params.pubBytes] - public key point encoded as bytes
+* @param {Array<Byte>} [params.privBytes] - private key point encoded as bytes
+* @param {Array<Byte>} [params.messagePrefix] - message prefix
+*
+* Any optional params not passed in, will be computed lazily (when possible)
+*/
+function KeyPair(eddsa, params) {
+  var self = this;
+  this.eddsa = eddsa;
+
+  // Set all params passed in as private members, others will be computed lazily
+  // where possible.
+  Object.keys(params).forEach(function(k) {
```

Comment:
Yikes, this plays badly with hidden classes. Let's always define all params and put the manually, shouldn't be that much of them.